File: examples/use_gdal.R
#! /usr/bin/env Rscript

library(rgdal)
library(optparse)

# parse the command-line options
options_list = list(optparse::make_option(c("--filename"), type = "character"))
opt_parser = optparse::OptionParser(option_list = options_list);
options =  optparse::parse_args(opt_parser);

# print the name of the image file
cat('\n', 'Image file:    ' , options$filename, '\n')

# load a GDAL dataset from the image file
dataset <- GDAL.open(options$filename)

# get the matrix of values in the dataset
dataMatrix <- getRasterData(dataset)

# display the pixel values in the data set
cat('\n', 'Pixel values:', '\n\n')
print(dataMatrix)

File: examples/use_gdal.py
#! /usr/bin/env python2

import argparse
from osgeo import gdal
from gdalconst import GA_ReadOnly

# parse the command line arguments
parser = argparse.ArgumentParser()
parser.add_argument('filename', help='file to probe')
args = parser.parse_args()

# echo back the first argument
print("Image file:       {}".format(args.filename))

# open the file as a GDAL data set
dataset = gdal.Open(args.filename, GA_ReadOnly)

# print some properties of the data set
print("Driver:           {}".format(dataset.GetDriver().LongName))
print("Extent in pixels: {} x {} ".format(dataset.RasterXSize, dataset.RasterYSize))
print("Number of bands:  {}".format(dataset.RasterCount))